File: api_nestjs_distributed_order_platform/src/presentation/payment.controller.ts
import { Body, Controller, Post } from "@nestjs/common";
import { CreatePaymentUseCase } from "src/application/use-case/payment/create-payment.use-case";
import { PaymentCreateDto } from "./http/dto/payment.dto";
import { toPaymentResponse } from "./http/mappers/payment.mapper";

@Controller('payments')
export class PaymentController {
    constructor(
        private readonly createPaymentUseCase: CreatePaymentUseCase,
    ) {}

    @Post()
    async createPayment(@Body() body: PaymentCreateDto) {
        const payment = await this.createPaymentUseCase.execute(body);
        return toPaymentResponse(payment);
    }
}

File: api_java_distributed_order_platform/src/main/java/com/project/order/presentation/http/PaymentController.java
package com.project.order.presentation.http;

import com.project.order.application.usecase.payment.CreatePaymentUseCase;
import com.project.order.presentation.http.dto.PaymentDto.CreatePaymentRequest;
import com.project.order.presentation.http.dto.PaymentDto.PaymentResponse;
import com.project.order.presentation.http.mapper.PaymentMapper;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/payments")
public class PaymentController {

    private final CreatePaymentUseCase createPaymentUseCase;

    public PaymentController(CreatePaymentUseCase createPaymentUseCase) {
        this.createPaymentUseCase = createPaymentUseCase;
    }

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public PaymentResponse create(@Valid @RequestBody CreatePaymentRequest body) {
        return PaymentMapper.toResponse(createPaymentUseCase.execute(body.order_id(), body.amount()));
    }
}
